bears/general/FilenameBear.py

```diff
@@ -66,6 +74,24 @@ def run(self, filename, file,
         try:
             new_name = self._naming_convention[file_naming_convention](
                 filename_without_extension)
+            messages = []
+
+            if new_name != filename_without_extension:
+                messages.append(
+                    'Filename does not follow {} naming-convention.'.format(
+                        file_naming_convention))
+
+            if not filename_without_extension.startswith(filename_prefix):
+                new_name = filename_prefix + new_name
+                messages.append(
+                    'Filename does not use the {!r} prefix.'.format(
+                        filename_prefix))
+
+            if not filename_without_extension.endswith(filename_suffix):
+                new_name = new_name + filename_suffix
+                messages.append(
+                    'Filename does not use the {!r} suffix.'.format(
+                        filename_suffix))
```

Comment:
ah no I didn't mean to put the whole stuff into the try-block, just moving `messages = []` *before* the try-block :D

The prefix/suffix stuff should come after the `try`-block :+1:

So
```python
        messages = []

        try:
            new_name = self._naming_convention[file_naming_convention](
                filename_without_extension)
        except KeyError:
            self.err('Invalid file-naming-convention provided: ' +
                     file_naming_convention)
            return

        if new_name != filename_without_extension:
            messages.append(
                'Filename does not follow {} naming-convention.'.format(
                    file_naming_convention))

        if not filename_without_extension.startswith(filename_prefix):
            new_name = filename_prefix + new_name
            messages.append(
                'Filename does not use the {!r} prefix.'.format(
                    filename_prefix))

        if not filename_without_extension.endswith(filename_suffix):
            new_name = new_name + filename_suffix
            messages.append(
                'Filename does not use the {!r} suffix.'.format(
                    filename_suffix))
```